ui/core/src/utils/fetch.ts

```diff
@@ -17,6 +17,13 @@
 export async function fetch(...args: Parameters<typeof global.fetch>) {
```

Comment:
Could update the comment if it seems helpful

Reply:
Yeah definitely, also you're probably right about how using `Error` would be sufficient for this PR. I do think there is functionality we can build on top of this though, where having UserFriendlyError will be helpful. If you search for `getUserFriendlyErrors` in our internal codebase, there are ideas there that can we possibly leverage in the future.